tests/shared/partition.sh

```diff
@@ -53,5 +53,5 @@ clean_up_partition() {
     path="$1"
     escaped_path=${path//$'/'/$'\/'}
     sed -i "/${escaped_path}/d" /etc/fstab
-    umount ${path} || true  # no problem if not mounted
+    umount -l ${path} || true  # no problem if not mounted
```

Comment:
This might mask some other issues, what about checking if path is mounted and unmount only in that case?

Reply:
I don't think it masks issues and I also don't think that checking if path is mounted will help us.
In case of `/dev/shm`, it is mounted but `umount` fails:
```
# umount /dev/shm
umount: /dev/shm: target is busy.
```
However `# umount -l /dev/shm` passes because it detach a filesystem from the file hierarchy now and clean up references as soon as it is not busy anymore.

In case of `umount` without any option, it doesn't detach a filesystem, we can't remount it with new options, and thus we can't perform test scenarios properly.

Reply:
I agree with the lazy umount. My point was, what if even lazy umount fails? This way you will hide the error, SSGTS will report that it expected rule to pass but got fail. If you would rather umount only in case the path is mounted then these kind of issues would be correctly detected by the SSGTS which would report that script failed to prepare environment.

Reply:
Aha, I think I get it now. Updated with check if the path is mounted